File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint/fp_sa4_helper_clean_call_site_only.py
"""FP: SA4 context-sensitivity sibling -- this file only exercises the
CLEAN call site (literal argument). A sinking call site elsewhere must not
leak taint into this one via a flow-insensitive/summary-conflated
resolution."""

import subprocess


def run_cmd(host):
    subprocess.run("ping -c 1 " + host, shell=True)


def clean_call():
    run_cmd("localhost")

File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint_js/fp_sa4_helper_clean_call_site_only.js
function lookupUser(id) {
    db.query('SELECT * FROM users WHERE id=' + id);
}

// SA4 context-sensitivity sibling: this file only exercises the CLEAN call
// site (literal argument) -- a sinking call site elsewhere must not leak
// taint into this one via a flow-insensitive/summary-conflated resolution.
app.get('/safe', (req, res) => {
    lookupUser('admin');
    res.send('ok');
});
